src/managed/Microsoft.DotNet.Build.Bundle/Bundler.cs

```diff
@@ -0,0 +1,212 @@
+            if (type == FileType.Assembly)
+            {
+                long misalignment = (bundle.Position % AssemblyAlignment);
+
+                if (misalignment != 0)
+                {
+                    long padding = AssemblyAlignment - misalignment;
+                    bundle.Position += padding;
+                }
+            }
+
+            file.Position = 0;
+            long startOffset = bundle.Position;
+            file.CopyTo(bundle);
+
+            return startOffset;
+        }
+
+        bool ShouldEmbed(string fileName)
+        {
+            if (fileName.Equals(HostName))
+            {
+                // The bundle starts with the host, so ignore it while embedding.
+                return false;
+            }
+
+            if (fileName.Equals(RuntimeConfigDevJson))
+            {
+                // Ignore the machine specific configuration file.
+                return false;
+            }
+
+            if (Path.GetExtension(fileName).ToLower().Equals(".pdb"))
+            {
+                return EmbedPDBs;
+            }
+
+            return true;
+        }
+
+        FileType InferType(string fileName, Stream file)
+        {
+            if (fileName.Equals(DepsJson))
+            {
+                return FileType.DepsJson;
+            }
+
+            if (fileName.Equals(RuntimeConfigJson))
+            {
+                return FileType.RuntimeConfigJson;
+            }
+
+            if (fileName.Equals(Application))
+            {
+                return FileType.Application;
+            }
+
+            try
+            {
+                PEReader peReader = new PEReader(file);
+                CorHeader corHeader = peReader.PEHeaders.CorHeader;
+                if ((corHeader != null) && ((corHeader.Flags & CorFlags.ILOnly) != 0))
+                {
+                    return FileType.Assembly;
+                }
+            }
+            catch (BadImageFormatException)
+            {
+            }
+
+            return FileType.Extract;
+        }
+
+        void GenerateBundle()
+        {
+            string bundlePath = Path.Combine(OutputDir, HostName);
+
+            if (File.Exists(bundlePath))
+            {
+                Program.Log($"Ovewriting existing File {bundlePath}");
+            }
+
+            // Start with a copy of the host executable.
+            // Copy the file to preserve its permissions.
+            File.Copy(Path.Combine(ContentDir, HostName), bundlePath, overwrite: true);
+
+            using (BinaryWriter writer = new BinaryWriter(File.OpenWrite(bundlePath)))
+            {
+                Stream bundle = writer.BaseStream;
+                Manifest manifest = new Manifest();
+
+                bundle.Position = bundle.Length;
+                string [] contents = Directory.GetFiles(ContentDir);
+                Array.Sort(contents); // Sort the file names to keep the bundle construction deterministic.
```

Comment:
This seems to use the current culture unless you pass in your own comparer, so you must pass in eg. `StringComparer.Ordinal` to make it deterministic.